go-update/src/test/java/org/reactome/release/goupdate/JavaVsPerlComparison.java

```diff
@@ -223,4 +328,31 @@ else if (javaGoInst.getSchemClass().getName().equals(ReactomeJavaConstants.GO_Mo
 		System.out.println(catActDiffCount.get() + " CatalystActivities had differences.");
 		System.out.println(catActSameCount.get() + " CatalystActivities were the same.\n");
 	}
+
+	private static String getAccession(GKInstance instance)
+	{
+		try {
+			return (String) instance.getAttributeValue(ReactomeJavaConstants.accession);
+		} catch (Exception e) {
+			throw new RuntimeException("Can't find accession for instance " + instance.getDBID(), e);
+		}
+	}
+
+	private static String getNameValuesAsString(Set<String> values)
+	{
+		return values
+			.stream()
```

Comment:
Are streams really necessary for this sort of operation?

Reply:
Fair point - I replaced this with an inline String.join() operation: https://github.com/reactome/data-release-pipeline/pull/121/commits/cf7396e26c9ffe9bf9acc8024dbddc376adc01d9

I tried to do the same for https://github.com/reactome/data-release-pipeline/pull/121/files/9a9fd71d2fd9cc06e60023dbba7fe18787e88547..01f3cf3fc99e30e9a14e697af6e22683f989212c#diff-a1b0b46e16fc2dac4e77db0ea04b1bb7R350 but since it's a set of Map entries, it needs to be mapped to String values before being joined and a stream seems like a more readable way to do that over a for-loop.

Reply:
Ok, just be aware that streams often incur some overhead (not always a trivial amount), and they are not always more readable.

Reply:
Understood!  I always try to make my code as easy to follow as possible as the first priority and then optimize any hot spots after that.  I've found the principle of "it's easier to optimize correct code rather than correct optimize code" is usually right.

Reply:
I'm not sure that "simply don't use the streams API unless necessary" is an optimization, so...